File: practicas/practica2/src/main/java/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 */
public class EnemyStarShip {
    private String name;
    private float ammoPower;
    private float shieldPower;
    private Loot loot;
    private Damage damage;

    EnemyStarShip(String name, float ammoPower, float shieldPower, Loot loot, Damage damage) {
        this.name = name;
        this.ammoPower = ammoPower;
        this.shieldPower = shieldPower;
        this.loot = loot;
        this.damage = damage;
    }

    EnemyStarShip(EnemyStarShip e) {
        this(e.getName(), e.getAmmoPower(), e.getShieldPower(), e.getLoot(), e.getDamage());
    }
    
    public float fire() {
        return ammoPower;
    }
    
    public float getAmmoPower() {
        return ammoPower;
    }

    public Damage getDamage() {
        return new Damage(damage);
    }

    public float getShieldPower() {
        return shieldPower;
    }
    
    public float protection() {
        return shieldPower;
    }
    
    /**
     * @param shot potencia del disparo
     * @return el resultado del disparo recibido, si el nivel de la protección de los escudos es menor que la intensidad del disparo,
     * la nave enemiga no resiste (DONOTRESIST).  En casocontrario resiste el disparo (RESIST).
     * 
     */
    public ShotResult recieveShot(float shot) {
        if(shot > getShieldPower()){
            return ShotResult.DONOTRESIST;
        }
        
        return ShotResult.RESIST;
    }

    public Loot getLoot() {
        return new Loot(loot.getNSupplies(), loot.getNWeapons(), loot.getNShields(), loot.getNHangars(), loot.getNMedals());
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "Name: " + name + "\nammoPower: " + ammoPower + "\nShieldPower: " + shieldPower + "\nLoot: " + loot.toString()
                + "\nDamage: " + damage.toString();
    }
    
    
    
}

File: practicas/practica3_rb/EnemyStarShip.rb
#encoding: utf-8

require_relative 'ShotResult'

module Deepspace

    class EnemyStarShip

        #Constructor con parametros
        def initialize(n, a, s, l, d)
            @name = n
            @ammoPower = a
            @shieldPower = s
            @loot = l
            @damage = d
        end

        def to_s
            "Name: " + @name + "\nammoPower: " + @ammoPower.to_s + "\nshieldPower: " + @shieldPower.to_s + "\nloot: " + @loot.to_s + "\ndamage: " + @damage.to_s + "\n"
        end
        #Constructor de copia
        def self.newCopy(e)
            newe = EnemyStarShip.new(e.name, e.ammoPower, e.shieldPower, e.loot, e.damage)
            return newe
        end

        #Consultores
        attr_reader :name
        attr_reader :ammoPower
        attr_reader :shieldPower
        attr_reader :loot
        attr_reader :damage

        def protection
            @shieldPower
        end

        def fire
            @ammoPower
        end

        def receiveShot(shot)
            if shieldPower < shot
                return ShotResult::DONOTRESIST
            else
                return ShotResult::RESIST
            end
        end

        def getUIversion()
            EnemyToUI.new(self)
        end


    end
end
